mne/viz/evoked.py

```diff
@@ -1198,3 +1199,437 @@ def plot_evoked_joint(evoked, times="peaks", title='', picks=None,
+
+    # second, color
+    # check: is color a list?
+    if (colors is not None and not isinstance(colors, string_types) and
+            not isinstance(colors, dict) and len(colors) > 1):
+        colors = dict((condition, color) for condition, color
+                      in zip(conditions, colors))
+
+    if not isinstance(colors, dict):  # default colors from M Waskom's Seaborn
+        # XXX should put a good list of default colors into defaults.py
+        colors_ = ['#e41a1c', '#377eb8', '#4daf4a', '#984ea3', '#ff7f00',
+                   '#1b9e77', '#d95f02', '#7570b3', '#e7298a', '#66a61e']
+        if len(conditions) > len(colors_):
+            msg = ("Trying to plot more than {0} conditions. We provide"
+                   "only {0} default colors. Please supply colors manually.")
+            raise ValueError(msg.format(len(colors_)))
+        colors = dict((condition, color) for condition, color
+                      in zip(conditions, colors_))
+    else:
+        colors = _setup_styles(conditions, colors, "color", "grey")
+
+    # third, linestyles
+    if not isinstance(linestyles, dict):
+        linestyles = dict((condition, linestyle) for condition, linestyle in
+                          zip(conditions, ['-'] * len(conditions)))
+    else:
+        linestyles = _setup_styles(conditions, linestyles, "linestyle", "-")
+
+    # fourth, put it all together
+    if styles is None:
+        styles = dict()
+    for condition, color, linestyle in zip(conditions, colors, linestyles):
+        styles[condition] = styles.get(condition, dict())
+        styles[condition]['c'] = styles[condition].get('c', colors[condition])
+        styles[condition]['linestyle'] = styles[condition].get(
+            'linestyle', linestyles[condition])
+    # We now have a 'styles' dict with one entry per condition, specifying at
+    # least color and linestyles.
+
+    # the actual plot
+    any_negative, any_positive = False, False
+    for condition in conditions:
+        # plot the actual data ('d') as a line
+        if ch_type == 'grad' and gfp is False:
+            d = ((_merge_grad_data(evokeds[condition]
+                 .data[picks, :])).T * scaling).mean(-1)
+        else:
+            func = np.std if gfp is True else np.mean
+            d = func((evokeds[condition].data[picks, :].T * scaling), -1)
+        axes.plot(times, d, zorder=1000, label=condition, **styles[condition])
+        if any(d > 0):
+            any_positive = True
+        if any(d < 0):
+            any_negative = True
+
+        # plot the confidence interval (standard error of the mean/'sem_')
+        if ci and gfp is not True:
+            sem_ = sem_array[condition]
+            axes.fill_between(times, sem_[0].flatten() * scaling,
+                              sem_[1].flatten() * scaling, zorder=100,
+                              color=styles[condition]['c'], alpha=.333)
+
+    # truncate the y axis ... yes, it's that complicated
```

Comment:
can you make this juggling of "despining" ylim dance a separate private function?